ReactiveUI.Tests/ObservableAsPropertyHelperTest.cs

```diff
@@ -132,4 +138,79 @@ public void ToPropertyShouldFireBothChangingAndChanged()
             Assert.Equal("Baz", resultChanged[1].Value);
         }
     }
+
+    class UnitTestMode : IModeDetector
+    {
+        public bool? InDesignMode()
+        {
+            return false;
+        }
+
+        public bool? InUnitTestRunner()
+        {
+            return true;
+        }
+    }
+
+    public class ToPropertyRaceConditionSpec : IDisposable
+    {
+        public ToPropertyRaceConditionSpec()
+        {
+            ModeDetector.OverrideModeDetector(null);
+        }
+
+        public void Dispose()
+        {
+            ModeDetector.OverrideModeDetector(new UnitTestMode());
+        }
+
+        public class Fixture : ReactiveObject
+        {
+            public ObservableAsPropertyHelper<bool> _A;
+
+            public bool A
+            {
+                get { return _A.Value; }
+            }
+        }
+
+        [Fact]
+        public void ToPropertyShouldSubscribeOnlyOnce()
+        {
+
+
+                var count = 0;
+                var obs = Observable.Create<bool>(observer =>
+                {
+                    count++;
+                    // We need to generate a value on subscription
+                    // which is different than the default value.
+                    // This triggers the property change firing
+                    // upon subscription in the ObservableAsPropertyHelper
+                    // constructor.
+                    observer.OnNext(true);
+                    return Disposable.Empty;
+                });
+
+                var f = new Fixture();
+
+                f._A = obs.ToProperty(f, t => t.A, false);
```

Comment:
Maybe prefer using the recommended way of declaring properties ? i.e. move it to the VM class and use:
```
Observable.Return(true).Do(_ => _count++).ToProperty(this, x => x.A, out _A);
```